### ISSUE TITLE
Offshore Wind Generation Output Seams and Striations

### ISSUE THREAD

**Issue by WilliamsTravis** [NREL/reV#182]
**Bug Description**
The mean lcoe and capacity factor outputs from reV generation for windpower in the United State have distinct patterns in their geographic distribution. There is a large seam between the east and west halves of the country and, on the east side of this seam, there are striations along the coasts and the Canadian border.

The resource outputs look good and this only happens when running the offshore module.

**Full Traceback**
No obvious errors found in the logs. This warning was present in collect: "CollectionWarning: GID indices for source file "wind_gen_2012_node16_offshore.h5" are not sequential in destination file!"

However the same warnings are present for offshore runs with a previous version of reV (0.3.1) without these patterns. 

**Screenshots**

![Screen Shot 2020-07-08 at 4 01 33 PM](https://user-images.githubusercontent.com/21287088/86974885-52cf5100-c134-11ea-9e5a-73b757c9a029.png)

![Screen Shot 2020-07-08 at 1 58 05 PM](https://user-images.githubusercontent.com/21287088/86966797-f9acf080-c126-11ea-988f-f95cf6ae4eb9.png)

**To Reproduce**
Recreate the reV pipeline in  '/projects/rev/new_projects/stripe_test/rev_045/wind'.

**Expected behavior**
A more gradual distribution of capacity factor and lcoe values across the study area.

**System (please complete the following information):**
 - OS: Linux
 - Compute hardware:  Eagle
 - Version: 0.4.5

**Comment by grantbuster** [comment]
@WilliamsTravis, I've got good news and I've got bad news. 

The good news is that the collected wind speed mean data looks fine. I think the offshore modules and collect are performing exactly as designed. That's great (especially for me!) because debugging the offshore module is often painful. 

The bad news is the weird CF and LCOE striations are observable in the chunked generation output files. For example, here are map plots from wind_gen_2012_node17.h5 with Florida visible in the lower section. You can see that these weird bands seem to be output directly by the SAM compute modules. I checked your project points and it looks like the offshore turbines are being mapped to the offshore sites appropriately so there shouldn't be any weirdness there. 

I'll have to think more about what could be going wrong in this.... @MRossol feel free to chime in. 

CF_MEAN from wind_gen_2012_node17.h5:
![image](https://user-images.githubusercontent.com/12700356/86980326-cbd4a580-c140-11ea-9a46-c518dba25a7f.png)


LCOE from wind_gen_2012_node17.h5:
![image](https://user-images.githubusercontent.com/12700356/86979885-a1ceb380-c13f-11ea-9554-81649b308c3c.png)

WS_MEAN from wind_gen_2012_node17.h5:
![image](https://user-images.githubusercontent.com/12700356/86980160-61236a00-c140-11ea-96da-609aab204d98.png)

**Comment by MRossol** [comment]
Super funky... Is it just me or does this appear to be isolated to the Western half of the US? I think the first thing to try is running a fixed SAM config for all of CONUS and see what we get...

**Comment by grantbuster** [comment]
There are still some issues on the west. See the california coast below. @WilliamsTravis, can we charge a few hours to look into this? 

![image](https://user-images.githubusercontent.com/12700356/87053123-d03fa380-c1be-11ea-8b39-8486068f0276.png)

**Comment by WilliamsTravis** [comment]
Hey, sure, use this one. Nick asked to let him know if it'll take more than a few hours though, it's getting low. 

**[OOSP.10291.17.09.01] SPIA Solar Pilot tn1**

**Comment by grantbuster** [comment]
I'm not sure how long it will take but it seems like a pretty bad bug. I was able to show a smooth cf_mean map with a single config in the project points but not with two configs. I added another test on the project points mapping but it seems to work fine. I'm a bit stumped.

**Comment by MRossol** [comment]
Any chance this is on the PySAM side? We haven't changed much to project points or generation since v0.3.1 besides updating the PySAM dependency...

**Comment by grantbuster** [comment]
I think not. Like I said, I was able to show a smooth cf_mean map with a single config so I dont think PySAM is doing anything weird. None of this is touching ORCA. The offshore site identification appears to be correct. I think it is an issue mapping the configs to the correct sites. It could be in [this commit](https://github.com/NREL/reV/commit/6a338421b94c0ed122bc33631b4b0e1d094e0681#diff-233159fb7d4df6d54463b40200c7ce7a). Feel free to help me look. I'm still confused.

**Comment by grantbuster** [comment]
I think i found it. It's something dumb.

**Comment by grantbuster** [comment]
@WilliamsTravis, bug is fixed in Pull Request #183. Will close and tag a new release when the tests finish.

The issue was that you're using old PySAM v1 keys for the wind farm losses. So you'll get warnings like this (/projects/rev/new_projects/stripe_test/rev_045/wind/logs/stdout/wind_gen_2012_17_3724979.o):

`WARNING - 2020-07-08 11:39:28,302 [version_checker.py:104] : It appears old SAM v1 keys are being used. Updated key "wind_farm_losses_percent" to "turb_generic_loss".`

This happens on a high level before parallelization. The code had a bug where the keys would get fixed but then the parallel workers would reload the sam config from disk. This only happens with project points that use multiple wind turbine configs. 

So in summary, this bug occurred for jobs that had:

1. multiple turbine configs
2. turbine configs had outdated SAM v1 keys
3. parallel execution

**Comment by grantbuster** [comment]
This bug is included in https://github.com/NREL/reV/releases/tag/v0.4.6

Also updated module files.
